### ISSUE TITLE
Style/GuardClause cop inspection error

### ISSUE THREAD

**Issue by peaky76** [rubocop/rubocop#11324]
```An error occurred while Style/GuardClause cop was inspecting <filename>```

Reporting the bug here as requested by the message

--------

## Expected behavior

Rubocop should flag any style issues in the file and then proceed to completion

## Actual behavior

Rubocop displays the `An error occurred while Style/GuardClause cop was inspecting <filename>` error, passes everything else, then ends with a Rubocop failed message. Full debug message below:

```
For /vagrant: configuration from /vagrant/.rubocop.yml
configuration from /home/vagrant/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/rubocop-rails-2.17.3/config/default.yml
configuration from /home/vagrant/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/rubocop-rails-2.17.3/config/default.yml
Default configuration from /home/vagrant/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/rubocop-1.40.0/config/default.yml
configuration from /home/vagrant/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/rubocop-minitest-0.25.0/config/default.yml
configuration from /home/vagrant/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/rubocop-minitest-0.25.0/config/default.yml
Inheriting configuration from /vagrant/.rubocop_todo.yml
Inheriting configuration from /vagrant/.rubocop_temp.yml
.rubocop.yml: Metrics/ClassLength:Exclude overrides the same parameter in .rubocop_temp.yml
Use parallel by default.
Running parallel inspection
Loading cache from /home/vagrant/.cache/rubocop_cache/2e904ce412c4df0d7e458cebb09e4499b3829656/6d7a3b621ca1730e04accd938619e4bdab66cfb1/1af3ef8102e208cef319f428d2b17ea62dc4136d
Loading cache from /home/vagrant/.cache/rubocop_cache/2e904ce412c4df0d7e458cebb09e4499b3829656/6d7a3b621ca1730e04accd938619e4bdab66cfb1/af80b5899e0b01465ab51a539ce5f975a708f9c9
Loading cache from /home/vagrant/.cache/rubocop_cache/2e904ce412c4df0d7e458cebb09e4499b3829656/6d7a3b621ca1730e04accd938619e4bdab66cfb1/5f20fe4a51f9ba311c412e41a49a062702e120b4
Loading cache from /home/vagrant/.cache/rubocop_cache/2e904ce412c4df0d7e458cebb09e4499b3829656/6d7a3b621ca1730e04accd938619e4bdab66cfb1/d7f8b87f19b0f1427f794a30ea84b8508f88a9a8
Loading cache from /home/vagrant/.cache/rubocop_cache/2e904ce412c4df0d7e458cebb09e4499b3829656/6d7a3b621ca1730e04accd938619e4bdab66cfb1/4422a9c34299e27ac6cd01993787d4bfcce7dbc2
Loading cache from /home/vagrant/.cache/rubocop_cache/2e904ce412c4df0d7e458cebb09e4499b3829656/6d7a3b621ca1730e04accd938619e4bdab66cfb1/fedf7f3b12f02bdaddb535a53600cd785bd64573
Loading cache from /home/vagrant/.cache/rubocop_cache/2e904ce412c4df0d7e458cebb09e4499b3829656/6d7a3b621ca1730e04accd938619e4bdab66cfb1/e3afbb828c5d2143e63b6f27612bd90dc2c2d3c5
Loading cache from /home/vagrant/.cache/rubocop_cache/2e904ce412c4df0d7e458cebb09e4499b3829656/6d7a3b621ca1730e04accd938619e4bdab66cfb1/6026320d1b79f9d17fdd1e1c2e4169247dad0db3
Loading cache from /home/vagrant/.cache/rubocop_cache/2e904ce412c4df0d7e458cebb09e4499b3829656/6d7a3b621ca1730e04accd938619e4bdab66cfb1/bc3b56cc7e798bcc7b720a032d01d10e115fa83e
Loading cache from /home/vagrant/.cache/rubocop_cache/2e904ce412c4df0d7e458cebb09e4499b3829656/6d7a3b621ca1730e04accd938619e4bdab66cfb1/bb4ef85fc32718fbc80bb54c3b950c7ef01f8c4c
Loading cache from /home/vagrant/.cache/rubocop_cache/2e904ce412c4df0d7e458cebb09e4499b3829656/6d7a3b621ca1730e04accd938619e4bdab66cfb1/08961aa303d8c2c63cb5d3f4cadebaf7ddfb4f1c
Loading cache from /home/vagrant/.cache/rubocop_cache/2e904ce412c4df0d7e458cebb09e4499b3829656/6d7a3b621ca1730e04accd938619e4bdab66cfb1/17f35766b47385d014f4ffdb0e116a40068b6fb7
Loading cache from /home/vagrant/.cache/rubocop_cache/2e904ce412c4df0d7e458cebb09e4499b3829656/6d7a3b621ca1730e04accd938619e4bdab66cfb1/d5d7c875bd7a5a0e5456632c478a7d8281ac2a84
Loading cache from /home/vagrant/.cache/rubocop_cache/2e904ce412c4df0d7e458cebb09e4499b3829656/6d7a3b621ca1730e04accd938619e4bdab66cfb1/799f7329ae88bcaebba01adaa34ef14fe19f66ba
Loading cache from /home/vagrant/.cache/rubocop_cache/2e904ce412c4df0d7e458cebb09e4499b3829656/6d7a3b621ca1730e04accd938619e4bdab66cfb1/d793e716ff20b58217e71881e6b6dfbea64675c6
Loading cache from /home/vagrant/.cache/rubocop_cache/2e904ce412c4df0d7e458cebb09e4499b3829656/6d7a3b621ca1730e04accd938619e4bdab66cfb1/b19f97fc5515a97b5e45f32f46e93bee65019287
An error occurred while Style/GuardClause cop was inspecting /vagrant/app/controllers/admin/question_catalogue/questions_controller.rb:134:6.
undefined method `heredoc?' for s(:send,
  s(:const, nil, :I18n), :t,
  s(:str, "admin.question_catalogue.questions.too_high_answer_count_max")):RuboCop::AST::SendNode
/home/vagrant/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/rubocop-1.40.0/lib/rubocop/cop/style/guard_clause.rb:190:in `autocorrect'
/home/vagrant/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/rubocop-1.40.0/lib/rubocop/cop/style/guard_clause.rb:179:in `block in register_offense'
/home/vagrant/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/rubocop-1.40.0/lib/rubocop/cop/base.rb:346:in `correct'
/home/vagrant/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/rubocop-1.40.0/lib/rubocop/cop/base.rb:127:in `add_offense'
/home/vagrant/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/rubocop-1.40.0/lib/rubocop/cop/style/guard_clause.rb:176:in `register_offense'
/home/vagrant/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/rubocop-1.40.0/lib/rubocop/cop/style/guard_clause.rb:149:in `check_ending_if'
/home/vagrant/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/rubocop-1.40.0/lib/rubocop/cop/style/guard_clause.rb:140:in `check_ending_body'
/home/vagrant/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/rubocop-1.40.0/lib/rubocop/cop/style/guard_clause.rb:109:in `on_def'
/home/vagrant/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/rubocop-1.40.0/lib/rubocop/cop/commissioner.rb:100:in `public_send'
/home/vagrant/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/rubocop-1.40.0/lib/rubocop/cop/commissioner.rb:100:in `block (2 levels) in trigger_responding_cops'
/home/vagrant/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/rubocop-1.40.0/lib/rubocop/cop/commissioner.rb:160:in `with_cop_error_handling'
/home/vagrant/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/rubocop-1.40.0/lib/rubocop/cop/commissioner.rb:99:in `block in trigger_responding_cops'
/home/vagrant/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/rubocop-1.40.0/lib/rubocop/cop/commissioner.rb:98:in `each'
/home/vagrant/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/rubocop-1.40.0/lib/rubocop/cop/commissioner.rb:98:in `trigger_responding_cops'
/home/vagrant/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/rubocop-1.40.0/lib/rubocop/cop/commissioner.rb:69:in `on_def'
/home/vagrant/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/rubocop-ast-1.24.0/lib/rubocop/ast/traversal.rb:137:in `block in on_dstr'
/home/vagrant/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/rubocop-ast-1.24.0/lib/rubocop/ast/traversal.rb:137:in `each'
/home/vagrant/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/rubocop-ast-1.24.0/lib/rubocop/ast/traversal.rb:137:in `on_dstr'
/home/vagrant/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/rubocop-1.40.0/lib/rubocop/cop/commissioner.rb:71:in `on_begin'
/home/vagrant/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/rubocop-ast-1.24.0/lib/rubocop/ast/traversal.rb:154:in `on_class'
/home/vagrant/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/rubocop-1.40.0/lib/rubocop/cop/commissioner.rb:71:in `on_class'
/home/vagrant/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/rubocop-ast-1.24.0/lib/rubocop/ast/traversal.rb:138:in `on_while'
/home/vagrant/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/rubocop-1.40.0/lib/rubocop/cop/commissioner.rb:71:in `on_module'
/home/vagrant/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/rubocop-ast-1.24.0/lib/rubocop/ast/traversal.rb:138:in `on_while'
/home/vagrant/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/rubocop-1.40.0/lib/rubocop/cop/commissioner.rb:71:in `on_module'
/home/vagrant/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/rubocop-ast-1.24.0/lib/rubocop/ast/traversal.rb:20:in `walk'
/home/vagrant/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/rubocop-1.40.0/lib/rubocop/cop/commissioner.rb:86:in `investigate'
/home/vagrant/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/rubocop-1.40.0/lib/rubocop/cop/team.rb:154:in `investigate_partial'
/home/vagrant/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/rubocop-1.40.0/lib/rubocop/cop/team.rb:82:in `investigate'
/home/vagrant/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/rubocop-1.40.0/lib/rubocop/runner.rb:315:in `inspect_file'
/home/vagrant/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/rubocop-1.40.0/lib/rubocop/runner.rb:259:in `block in do_inspection_loop'
/home/vagrant/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/rubocop-1.40.0/lib/rubocop/runner.rb:293:in `block in iterate_until_no_changes'
/home/vagrant/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/rubocop-1.40.0/lib/rubocop/runner.rb:286:in `loop'
/home/vagrant/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/rubocop-1.40.0/lib/rubocop/runner.rb:286:in `iterate_until_no_changes'
/home/vagrant/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/rubocop-1.40.0/lib/rubocop/runner.rb:255:in `do_inspection_loop'
/home/vagrant/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/rubocop-1.40.0/lib/rubocop/runner.rb:138:in `block in file_offenses'
/home/vagrant/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/rubocop-1.40.0/lib/rubocop/runner.rb:163:in `file_offense_cache'
/home/vagrant/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/rubocop-1.40.0/lib/rubocop/runner.rb:137:in `file_offenses'
/home/vagrant/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/rubocop-1.40.0/lib/rubocop/runner.rb:73:in `block in warm_cache'
/home/vagrant/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/parallel-1.22.1/lib/parallel.rb:587:in `call_with_index'
/home/vagrant/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/parallel-1.22.1/lib/parallel.rb:557:in `process_incoming_jobs'
/home/vagrant/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/parallel-1.22.1/lib/parallel.rb:537:in `block in worker'
/home/vagrant/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/parallel-1.22.1/lib/parallel.rb:528:in `fork'
/home/vagrant/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/parallel-1.22.1/lib/parallel.rb:528:in `worker'
/home/vagrant/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/parallel-1.22.1/lib/parallel.rb:519:in `block in create_workers'
/home/vagrant/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/parallel-1.22.1/lib/parallel.rb:518:in `each'
/home/vagrant/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/parallel-1.22.1/lib/parallel.rb:518:in `each_with_index'
/home/vagrant/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/parallel-1.22.1/lib/parallel.rb:518:in `create_workers'
/home/vagrant/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/parallel-1.22.1/lib/parallel.rb:457:in `work_in_processes'
/home/vagrant/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/parallel-1.22.1/lib/parallel.rb:294:in `map'
/home/vagrant/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/parallel-1.22.1/lib/parallel.rb:238:in `each'
/home/vagrant/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/rubocop-1.40.0/lib/rubocop/runner.rb:73:in `warm_cache'
/home/vagrant/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/rubocop-1.40.0/lib/rubocop/runner.rb:46:in `run'
/home/vagrant/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/rubocop-1.40.0/lib/rubocop/cli/command/execute_runner.rb:26:in `block in execute_runner'
/home/vagrant/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/rubocop-1.40.0/lib/rubocop/cli/command/execute_runner.rb:52:in `with_redirect'
/home/vagrant/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/rubocop-1.40.0/lib/rubocop/cli/command/execute_runner.rb:25:in `execute_runner'
/home/vagrant/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/rubocop-1.40.0/lib/rubocop/cli/command/execute_runner.rb:17:in `run'
/home/vagrant/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/rubocop-1.40.0/lib/rubocop/cli/command.rb:11:in `run'
/home/vagrant/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/rubocop-1.40.0/lib/rubocop/cli/environment.rb:18:in `run'
/home/vagrant/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/rubocop-1.40.0/lib/rubocop/cli.rb:72:in `run_command'
/home/vagrant/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/rubocop-1.40.0/lib/rubocop/cli.rb:79:in `execute_runners'
/home/vagrant/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/rubocop-1.40.0/lib/rubocop/cli.rb:48:in `run'
/home/vagrant/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/rubocop-1.40.0/exe/rubocop:19:in `block in <top (required)>'
/home/vagrant/.rbenv/versions/3.0.4/lib/ruby/3.0.0/benchmark.rb:308:in `realtime'
/home/vagrant/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/rubocop-1.40.0/exe/rubocop:19:in `<top (required)>'
/home/vagrant/.rbenv/versions/3.0.4/bin/rubocop:25:in `load'
/home/vagrant/.rbenv/versions/3.0.4/bin/rubocop:25:in `<main>'
```

## Steps to reproduce the problem

The error is occurring in this private method in a controller file.

```      
def answer_count_warnings(limit)
      optional_answer_count = @question.optional_answers.count

      if limit.max.zero? && optional_answer_count.positive?
        flash[:warning] = I18n.t('admin.question_catalogue.questions.zero_answer_count_max')
      end

      if limit.max > optional_answer_count
        flash[:info] = I18n.t('admin.question_catalogue.questions.too_high_answer_count_max')
      end
end
```

If the call to this function is replaced with the function body, then all proceeds as normal (although it causes a complexity cop).
If the I18n is replaced by a plain string it also proceeds as normal. My workaround has therefore been to refactor the function like this:

```
 TOO_HIGH_MAX_WARNING = I18n.t('admin.question_catalogue.questions.too_high_answer_count_max')
 ZERO_MAX_WARNING = I18n.t('admin.question_catalogue.questions.zero_answer_count_max')
     
 ...
     
def answer_count_warnings(limit)
        return unless limit&.max

        optional_answer_count = @question.optional_answers.count
        flash[:warning] = ZERO_MAX_WARNING if limit.max.zero? && optional_answer_count.positive?
        flash[:info] = TOO_HIGH_MAX_WARNING if limit.max > optional_answer_count
end
```

## RuboCop version

```
1.40.0 (using Parser 3.1.3.0, rubocop-ast 1.24.0, running on ruby 3.0.4) [x86_64-linux]
  - rubocop-minitest 0.25.0
  - rubocop-rails 2.17.3

```

**Comment by koic** [comment]
This issue is a dup and it has been solved by https://github.com/rubocop/rubocop/pull/11250. Please upgrade to the latest RuboCop 1.14.1.
